rollup/core/classes/RollupLogger.cls

```diff
@@ -81,17 +112,59 @@ public virtual class RollupLogger extends Rollup implements ILogger {
   }
 
   private static ILogger getRollupLogger() {
-    ILogger loggerInstance;
-    if (String.isNotBlank(SELF.rollupControl.RollupLoggerName__c)) {
+    List<RollupLoggerPlugin__mdt> plugins = [SELECT DeveloperName FROM RollupLoggerPlugin__mdt WHERE RollupControl__c = :SELF.rollupControl.Id];
+    if (pluginMocks != null) {
+      plugins.addAll(pluginMocks);
+    }
+    return combineLoggers(plugins);
+  }
+
+  private static ILogger combineLoggers(List<RollupLoggerPlugin__mdt> plugins) {
+    ILogger rollLogger;
+    if (plugins.isEmpty()) {
+      return SELF;
+    }
+    RollupLoggerPlugin__mdt firstLogger = plugins.remove(0);
+    if (firstLogger.DeveloperName == SELF.getTypeName()) {
+      rollLogger = SELF;
+    } else {
+
       try {
-        loggerInstance = (ILogger) Type.forName(SELF.rollupControl.RollupLoggerName__c).newInstance();
-      } catch (Exception ex) {
-        SELF.log('cast to Rollup.ILogger failed with message: ' + ex.getMessage() + ', falling back to default logger', SELF, LoggingLevel.WARN);
-        loggerInstance = SELF;
+        rollLogger = (ILogger) Type.forName(firstLogger.DeveloperName).newInstance();
+      } catch(Exception ex) {
+        SELF.log('cast to ILogger failed with message: ' + ex.getMessage() + ', falling back to default logger', SELF, LoggingLevel.WARN);
+        rollLogger = SELF;
       }
-    } else {
-      loggerInstance = SELF;
+      if (plugins.isEmpty() == false) {
+        rollLogger = new CombinedLogger(rollLogger, combineLoggers(plugins));
```

Comment:
Huzzah, recursion! 🔁

Reply:
taken straight from the same exact pattern in `RollupEvaluator`